tests/framework/matcher/phase_test.go

```diff
@@ -1,10 +1,17 @@
 package matcher
 
 import (
+	"fmt"
+
 	. "github.com/onsi/ginkgo"
 	"github.com/onsi/ginkgo/extensions/table"
 	. "github.com/onsi/gomega"
 	v1 "k8s.io/api/core/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v13 "kubevirt.io/api/core/v1"
+	"kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
```

Comment:
It is odd to see CDI in e2e infrastructure from kubevirt/kubevirt. Even if it is in tests (which is odd by itself to see unit tests under the e2e tests folder).

I think it would have been nicer and cleaner to exercise this against kubevirt/kubevirt API.

Reply:
I will disagree. See https://github.com/kubevirt/kubevirt/blob/74a42a02f38f32299f55aae71f89d53b68344516/tests/storage/datavolume.go as an example where we also use `cdi`. Also, we have tide integration with datavolumes as we are embedding it to our api.

Reply:
You have linked to storage sub-package, which is focused on storage.
Here we have a "framework" or library that is suppose to be independent and let others use it without also importing CDI stuff as a side-effect.

Reply:
I think we broke the assumption long age, see https://github.com/kubevirt/kubevirt/blob/74a42a02f38f32299f55aae71f89d53b68344516/tests/framework/matcher/getter.go#L13 (Or maybe I am missing something)

Reply:
We never had any constraints like @EdDev mentioned. All good.

Reply:
It is a good time to start having constraints on dependencies.
If you think it is ok to have such dependencies, I will be glad to hear a good reasoning for it.

Reply:
Let's discuss this in another place as I don't think it fits here.